src/Razor/src/Microsoft.NET.Sdk.Razor/ReferenceResolver.cs

```diff
@@ -15,155 +17,138 @@ namespace Microsoft.AspNetCore.Razor.Tasks
     public class ReferenceResolver
     {
         private readonly HashSet<string> _mvcAssemblies;
-        private readonly Dictionary<string, ClassifiedAssemblyItem> _lookup = new Dictionary<string, ClassifiedAssemblyItem>(StringComparer.Ordinal);
+        private readonly IReadOnlyList<AssemblyItem> _assemblyItems;
+        private readonly Dictionary<AssemblyItem, Classification> _classifications = new Dictionary<AssemblyItem, Classification>();
 
         public ReferenceResolver(IReadOnlyList<string> targetAssemblies, IReadOnlyList<AssemblyItem> assemblyItems)
         {
             _mvcAssemblies = new HashSet<string>(targetAssemblies, StringComparer.Ordinal);
-
+            _assemblyItems = assemblyItems;
             foreach (var item in assemblyItems)
             {
-                var classifiedItem = new ClassifiedAssemblyItem(item);
-                _lookup[item.AssemblyName] = classifiedItem;
+                Lookup[item.AssemblyName] = item;
```

Comment:
Are there ever any cases when we have same assembly names different versions being included here? Of course, this question doesn't matter if the AssemblyName is the fully qualified variant.

I ask because I'd hate for a user to accidentally reference older and newer assemblies and then have the older assemblies trump the newer ones.

Reply:
These are fusion names (fully qualified) so we don't have that problem here. The only bit I wasn't sure was the same assembly appearing twice which is why this avoids `ToDictionary`